cli/src/main.rs

```diff
@@ -41,44 +42,102 @@ struct Args {
         output_file: Option<PathBuf>,
     },
     /// Get the status of a job in progress.
-    #[clap(display_order = 2)]
+    #[clap(display_order = 3)]
     Status { id: Uuid },
     /// Get results for a job run, given its ID.
-    #[clap(display_order = 3)]
+    #[clap(display_order = 4)]
     Results { id: Uuid },
     /// Get full job run history from the running process.
-    #[clap(display_order = 4)]
+    #[clap(display_order = 5)]
     History,
+    /// Liveness check: ping at least one node on the mesh.
+    Ping,
 }
 
 static SERVAL_NODE_URL: Mutex<Option<String>> = Mutex::new(None);
 
 /// Convenience function to build urls repeatably.
-fn build_url(path: String) -> String {
+fn build_url(path: String, version: Option<&str>) -> String {
     let baseurl = SERVAL_NODE_URL.lock().unwrap();
     let baseurl = baseurl
         .as_ref()
         .expect("build_url called while SERVAL_NODE_URL is None");
-    format!("{baseurl}/v1/{path}")
+    if let Some(v) = version {
+        format!("{baseurl}/v{v}/{path}")
+    } else {
+        format!("{baseurl}/{path}")
+    }
+}
+
+fn upload_manifest(manifest_path: PathBuf) -> Result<()> {
+    println!("Reading manifest: {}", manifest_path.display());
+    let manifest = Manifest::from_file(&manifest_path)?;
+
+    println!("Reading WASM executable:{}", manifest.binary().display());
+    let executable = read_file(manifest.binary().to_path_buf())?;
+
+    let client = reqwest::blocking::Client::builder()
+        .timeout(Duration::from_secs(60))
+        .build()?;
+
+    // Start building pretty output now that we're past the most likely errors.
+    println!();
+    let mut table = Table::new();
+    table.set_format(*prettytable::format::consts::FORMAT_CLEAN);
+    table.add_row(row!["WASM task name:", manifest.fq_name()]);
+    table.add_row(row!["Version:", manifest.version()]);
+
+    let url = build_url("storage/manifests".to_string(), Some("1"));
+    let response = client.post(url).body(manifest.to_string()).send()?;
+
+    if !response.status().is_success() {
+        table.add_row(row!["Storing the WASM manifest failed!".bold()]);
+        table.add_row(row![format!("{} {}", response.status(), response.text()?)]);
+        println!("{table}");
+        return Ok(());
+    }
+
+    let manifest_integrity = response.text()?;
+    table.add_row(row!["Manifest integrity:", manifest_integrity]);
+
+    let vstring = format!(
+        "storage/manifests/{}/executable/{}",
+        manifest.fq_name(),
+        manifest.version()
+    );
+    let url = build_url(vstring, Some("1"));
+    let response = client.put(url).body(executable).send()?;
+    if response.status().is_success() {
+        let wasm_integrity = response.text()?;
+        table.add_row(row!["WASM integrity:", wasm_integrity]);
+        table.add_row(row![
+            "To run:",
+            format!("cargo run -- run {}", manifest.fq_name()).bold().blue()
```

Comment:
Consider making this `cargo run -p serval`, which will work correctly whether you're in the `cli` subdirectory or not.